ARC-598: Update README and SUPPORT files. (#786)
atlassian/github-for-jira: SUPPORT.md

| Status   | Definition                 |
|----------|----------------------------|
| PENDING  | The sync has not started.  |

> | IN PROGRESS   | The sync has started and is still in progress. No historical information will be displayed in Jira, but the integration will work for new data sent in. |

Reviewer: That's not necessarily true. Especially if a sync takes a long time, *some* historical data will be in Jira already, even if it's still `IN PROGRESS`.

Maybe write something like: "The backfilling of historical data is in progress. Over time, more and more data will show up in Jira."

> | COMPLETE | The sync has finished. Historical Information will be displayed in Jira. |

Reviewer: ```suggestion
| COMPLETE | The sync has finished. Historical data will be displayed in Jira. |
```

| FAILED   | The sync hit an error and stopped without completing. Partial historical information may appear in Jira. |

> The time it takes to complete the sync will depend on the size of your installation. Since the sync scans branches, commits and pull requests for every repository in your installation, be mindful that selecting "All Repositories" will perform a scan of every repository in your account, including forks. If you have repositories with hundreds of thousands of forks (e.g. a fork of the Linux repo), the scan might take several hours to complete.

Reviewer: ```suggestion
The time it takes to complete the sync will depend on the size of your GitHub organization, especially the number of commits in your repositories. Since the sync scans branches, commits and pull requests for every repository in your installation, be mindful that selecting "All Repositories" will perform a scan of every repository in your account, including forks. If your repositories have a lot of commits, the process can take hours or even days to complete.
```

Reviewer: ```suggestion
The time it takes to complete the sync will depend on the size of your installation. Since the sync scans branches, commits and pull requests for every repository in your installation, be mindful that selecting "All Repositories" will perform a scan of every repository in your account, including forks. If you have repositories with hundreds of thousands of forks (e.g. a fork of the Linux repo), the scan might take several hours to complete.
```

Reviewer: ```suggestion
The time it takes to complete the sync will depend on the size of your GitHub organization, especially the number of commits in your repositories. Since the sync scans branches, commits and pull requests for every repository in your installation, be mindful that selecting "All Repositories" will perform a scan of every repository in your account, including forks. If your repositories have a lot of commits, the process can take hours or even days to complete.
```

This warning is shown when a branch or commit includes more than 100 issue keys. When a branch or commit exceeds this limit, we only send the first 100. This is enforced by Jira. This doesn't impact branches or commits that are under the limit or impact the sync status.

Still having trouble? Look up your problem in [Issues](https://github.com/atlassian/github-for-jira/issues) or [Create a new Issue](https://github.com/atlassian/github-for-jira/issues/new) if it doesn't already exist.

> ## Sync is STALLED

Author: We don't have STALLED anymore

For more information, check out [Using the integration](https://github.com/atlassian/github-for-jira#using-the-integration).

## Workflow transitions are not running

In order for transitions to run from your Smart Commit syntax, a permission requirement must be met.

The email address used for the commit in GitHub that has the transition in the commit message must match an email address that has permission to run the transition in your Jira instance.

You can check the email address on GitHub by adding `.patch` to the end of a commit URL. For example:

`https://github.com/atom/atom/commit/834f8f3d73d84e98a423fe0720abd833d5ca7a87.patch`

## My deployments are showing up as unmapped

> The app does a mapping between both GitHub and Jira deployments environments names. GitHub does not have a concept of "environment type" and users can name their environments whatever they like. Jira, on the other hand, expects the types of environment, in addition to the environment name. Those can only be development, testing, staging and production. We map GitHub's environments names to one of the Jira environment types. If there's no match, Jira will consider it unmapped. The mapping logic can be found [here](https://github.com/atlassian/github-for-jira/blob/main/src/transforms/deployment.ts#L34), but it simply considers [a set of common values](https://github.com/atlassian/github-for-jira/blob/main/src/transforms/deployment.ts#L49) and their variations, e.g. "prod-east" and "prod-west" are considered variants of "prod".

Reviewer: ```suggestion
The app does a mapping from GitHub environments to Jira deployment environments. GitHub does not have a concept of "environment type" and users can name their environments whatever they like. 

Jira, on the other hand, expects the environment type to be one of:

 * `development`, 
 * `testing`, 
 * `staging`, or
 * `production`. 
 
 We map GitHub's environments names to one of the Jira environment types. If there's no match, Jira will consider it unmapped. The mapping logic can be found [here](https://github.com/atlassian/github-for-jira/blob/main/src/transforms/deployment.ts#L34), but it simply considers [a set of common values](https://github.com/atlassian/github-for-jira/blob/main/src/transforms/deployment.ts#L49) and their variations, e.g. "prod-east" and "prod-west" are considered variants of "prod".
```
